dev-aws/kafka-shared/customer-billing.tf

```diff
@@ -23,8 +23,30 @@ module "invoice_fulfillment_producer" {
 module "bills_total_api_consumer" {
   source = "../../modules/consumer"
 
-  topic = kafka_topic.invoice_fulfillment.name
+  topic          = kafka_topic.invoice_fulfillment.name
   consumer_group = "bills-total-api-reader"
 
-  cert_common_name = "customer-billing/bills-total-api" 
+  cert_common_name = "customer-billing/bills-total-api"
+}
+
+resource "kafka_topic" "invoice_fulfillment_deadletter" {
+  name               = "bex.internal.accountreadytobefulfilled_deadletter"
+  replication_factor = 3
+  partitions         = 10
+  config = {
+    # keep data for 14 days
+    "retention.ms" = "1209600000"
+    # allow max 1 MB for a message
+    "max.message.bytes" = "1048576"
+    "compression.type"  = "zstd"
+    "cleanup.policy"    = "delete"
+  }
+}
+
+module "invoice_fulfillment_producer" {
```

Comment:
This has the same name as the module in l.15. I don't know how this reacts to duplicated module names.

I don't know if we want to have 1 module per service and we can combine the two somehow or if we want 1 module per (service, topic).

@sbuliarca Do you have an opinion on that?

Reply:
Looking at this (https://github.com/utilitywarehouse/kafka-cluster-config/blob/main/modules/producer/variables.tf#L2) we probably can only configure 1 topic per module.

Reply:
>This has the same name as the module in l.15. I don't know how this reacts to duplicated module names.

I don't think it allows using the same name.

>I don't know if we want to have 1 module per service and we can combine the two somehow or if we want 1 module per (service, topic).

Unfortunately the TF plugin we're using https://github.com/Mongey/terraform-provider-kafka doesn't allow us to combine then, so I'd say 1 module per (service, topic) is our option.
I'd name this something like `invoice_fulfillment_deadletter_producer`

Reply:
Thanks